server/datastore/mysql/mysql.go

```diff
@@ -283,8 +287,20 @@ func (ds *Datastore) writeChanLoop() {
 	}
 }
 
+var tracedDriverName string
+
+func init() {
+	var err error
+	tracedDriverName, err = otelsql.Register("mysql", semconv.DBSystemMySQL.Value.AsString())
+	apmsql.Register(tracedDriverName, &mysql.MySQLDriver{}, apmsql.WithDSNParser(apmmysql.ParseDSN))
+	tracedDriverName = "apm/mysql"
+	if err != nil {
+		panic(err)
+	}
+}
+
 func newDB(conf *config.MysqlConfig, opts *dbOptions) (*sqlx.DB, error) {
-	driverName := "mysql"
+	driverName := tracedDriverName
```

Comment:
Could we follow the same logic as we do with `mysql-mw` and make sure to default to the mysql driver when nothing is configured?

I know this is annoying, but it feels like we'd be following the same pattern in the code (which is good practice on its own) and also removing the moving parts we've got when people don't care about these metrics.

Reply:
The main issue with doing that is that we aren't given the relavent config here as far as I can tell.

Reply:
We are passed `config.MysqlConfig` when we need the logging config.

Reply:
Yup, which we can expand, or we can add the fleet config param to `newDB`.

Reply:
Its not simple to just add the fleet config to newDB since a separate config.Mysql object is used for the read replica config. And expanding will probably break stuff since our tests magically enumerate the config data structure.

What do you think is a good path forward?

Reply:
How about following the same path we do for `opts.interceptor` in that case?

Reply:
That would work, and actually APM takes over much of the work of interceptor it seems. Should we just replace interceptor with APM?

Reply:
> Should we just replace interceptor with APM?

Hm, the interceptor is meant to be ran locally for devs, APM feels much bulkier.